File: examples/python-client/check_live_flags.py
#!/usr/bin/env python3
"""Check live_enabled flags on all 4 NexaRail modules."""

import nexarail_client as nxc


def main():
    print("=== NexaRail Devnet — Live Flag Check (Python) ===\n")

    modules = ["settlement", "escrow", "treasury", "payout"]
    results = {}

    for mod in modules:
        try:
            params = nxc.get_params(mod)
            live = params.get("live_enabled")
            extra = {}
            if mod == "settlement" and params:
                extra["treasury_routing"] = params.get("treasury_routing_enabled")
                extra["burn_routing"] = params.get("burn_routing_enabled")
            results[mod] = {"live": live, **extra}

            line = f"  {mod:<12} live_enabled: {live}"
            if mod == "settlement":
                line += (
                    f"  treasury_routing: {extra.get('treasury_routing', '?')}"
                    f"  burn_routing: {extra.get('burn_routing', '?')}"
                )
            print(line)
        except Exception as e:
            results[mod] = {"error": str(e)}
            print(f"  {mod:<12} ERROR: {e}")

    # Summary: all should be false (devnet not live)
    pass_ = True
    count = 0
    for mod, r in results.items():
        if "error" in r:
            pass_ = False
            continue
        if r.get("live") is True:
            pass_ = False
        count += 1

    print("")
    print(f"  Summary: {'PASS' if pass_ else 'FAIL'} ({count}/{len(modules)} modules checked)")
    if not pass_:
        flagged = [m for m, r in results.items() if r.get("live") is True]
        if flagged:
            print(f"  Flagged as live: {', '.join(flagged)}")

    # Also check node status for context
    try:
        ns = nxc.node_status()
        print(f"\n  Node: height={ns.get('sync_info', {}).get('latest_block_height', '?')}  "
              f"chain={ns.get('node_info', {}).get('network', '?')}")
    except Exception:
        print("\n  Node status: unreachable")


if __name__ == "__main__":
    main()

File: examples/node-client/src/checkLiveFlags.js
import { getParams, nodeStatus } from './client.js';

async function checkFlags() {
  console.log('=== NexaRail Devnet — Live Flag Check (Node.js) ===\n');

  const modules = ['settlement', 'escrow', 'treasury', 'payout'];
  const results = {};

  for (const mod of modules) {
    try {
      const params = await getParams(mod);
      const live = params?.live_enabled;
      const extra = {};
      if (params && mod === 'settlement') {
        extra.treasury_routing = params.treasury_routing_enabled;
        extra.burn_routing = params.burn_routing_enabled;
      }
      results[mod] = { live, ...extra };
      process.stdout.write(`  ${mod.padEnd(12)} live_enabled: ${String(live ?? 'MISSING')}`);
      if (mod === 'settlement') {
        process.stdout.write(`  treasury_routing: ${String(extra.treasury_routing ?? '?')}  burn_routing: ${String(extra.burn_routing ?? '?')}`);
      }
      process.stdout.write('\n');
    } catch (err) {
      results[mod] = { error: err.message };
      console.log(`  ${mod.padEnd(12)} ERROR: ${err.message}`);
    }
  }

  // Summary: all should be false (devnet not live)
  let pass = true;
  let count = 0;
  for (const [mod, r] of Object.entries(results)) {
    if (r.error) { pass = false; continue; }
    if (r.live === true) { pass = false; }
    count++;
  }

  console.log('');
  console.log(`  Summary: ${pass ? 'PASS' : 'FAIL'} (${count}/${Object.keys(results).length} modules checked)`);
  if (!pass) {
    const flagged = Object.entries(results).filter(([, r]) => r.live === true).map(([m]) => m);
    if (flagged.length) console.log(`  Flagged as live: ${flagged.join(', ')}`);
  }

  // Also check node status for context
  try {
    const ns = await nodeStatus();
    console.log(`\n  Node: height=${ns.sync_info?.latest_block_height ?? '?'}  chain=${ns.node_info?.network ?? '?'}`);
  } catch {
    console.log('\n  Node status: unreachable');
  }
}

checkFlags().catch(err => {
  console.error('checkLiveFlags failed:', err);
  process.exit(1);
});
